### PR TITLE
!status

### PR DESCRIPTION
**Pull request by LW001**
RE #27 

As described in https://trello.com/c/Do2gYoM4/.

Using the command works the following way:
`!status <ID/LINK> <NEW STATUS> <COMMENT>`

ID/LINK works using the magical UVRegex(TM)
NEW STATUS accepts the following values:
review, re, under review, ur, u, planned, plan, pl, p, started, start, completed, complete, co, c, declined, decline, de, d
COMMENT accepts a comment to be added, also filtering out pipes.
UV endpoint: https://developer.uservoice.com/docs/api/reference/#_api_v1_forums_forum_id_suggestions_suggestion_id_respond_put

### CODE REVIEW

File: Commands/Commands/uservoice.js [hunk 1]
```diff
@@ -129,6 +129,83 @@ commands.submit = {
...
+        let statncontent = s.join(' ');
+        let grabstatusidfromhere = statncontent.match(grabidregex);
+        let statusname = grabstatusidfromhere[2];
+        let statusid = null;
+        if (statusname.match(/(review|re|r|under review|ur|u|2)/)) { statusid = '1707873'; statusname = 'Under Review'; }
+        else if (statusname.match(/(planned|plan|pl|p|3)/)) { statusid = '1707876'; statusname = 'Planned'; }
+        else if (statusname.match(/(started|start|st|s|4)/)) { statusid = '1707879'; statusname = 'Started'; }
+        else if (statusname.match(/(completed|complete|co|c|5)/)) { statusid = '1707882'; statusname = 'Completed'; }
+        else if (statusname.match(/(declined|decline|de|d|6)/)) { statusid = '1707885'; statusname = 'Declined'; }
+        else if (statusname.match(/(none|x|1)/)) { statusid = null; statusname = 'None'; }
```
Comment (nullpixel) [inline Commands/Commands/uservoice.js]: this if statement is a bit eh, could this be refined?

File: Commands/Commands/uservoice.js [hunk 2]
```diff
@@ -129,6 +129,83 @@ commands.submit = {
...
+        let parts = idt.match(UVRegex)
+        let id
+        if (parts === null) {
```
Comment (nullpixel) [inline Commands/Commands/uservoice.js]: make this `id = parts === null ? idt : parts[2]` or `id = (parts && parts[2]) || idt`

File: Commands/Commands/uservoice.js [hunk 3]
```diff
@@ -129,6 +129,83 @@ commands.submit = {
...
+          id = idt
+        } else {
+          id = parts[2]
+        }
+        let content = grabstatusidfromhere[1]
```
Comment (nullpixel) [inline Commands/Commands/uservoice.js]: this variable name is too verbose. Use camel case.

File: Commands/Commands/uservoice.js [hunk 4]
```diff
@@ -129,6 +129,83 @@ commands.submit = {
...
+        if (content.startsWith("|")) content = content.slice(1).trim()
+        c.put(`forums/${config.uservoice.forumId}/suggestions/${id}/respond.json`, {
+          response: {
+            status_id: statusid,
+            text: content
+          }
+        }).then(data => {
+          let uvurl = `https://${config.uservoice.subdomain}.${config.uservoice.domain}/forums/${config.uservoice.forumId}/suggestions/${id}`
+          msg.reply('The suggestion ' + uvurl + ' has been successfully updated to `' + statusname + '`').then(scssmsg => {
```
Comment (nullpixel) [inline Commands/Commands/uservoice.js]: picky, but use backticks and ${}

**Review by nullpixel — CHANGES_REQUESTED** [review]
picky, but will improve code

**Comment by Dougley** [comment]
Closing this, superseded by #35

**Pull request closed without merging**